packages/mobile/components/ActivityList.svelte

```diff
@@ -1,44 +1,92 @@
+        title: string | undefined
+        amount: number | undefined
+        activity: Activity
+    }
 
     export let onTileClick: (activity: Activity) => unknown = () => {}
     export let onReject: (activity: Activity) => unknown = () => {}
     export let onClaim: (activity: Activity) => unknown = () => {}
 
-    $: setAsyncStatusOfAccountActivities($time)
+    let searchValue = ''
 
-    $: activityListWithTitles = $queriedActivities.map((activity, index) => {
-        const currentTitle = getActivityGroupTitleForTimestamp(activity.time)
-        const previousTitle = $queriedActivities[index - 1]
-            ? getActivityGroupTitleForTimestamp($queriedActivities[index - 1]?.time)
-            : undefined
-        if (currentTitle !== previousTitle || index === 0) {
-            const amount = $queriedActivities.filter(
-                (activity) => getActivityGroupTitleForTimestamp(activity.time) === currentTitle
-            ).length
-
-            return { title: currentTitle, amount, activity }
-        } else {
-            return { title: undefined, amount: undefined, activity }
-        }
-    })
+    $: setAsyncStatusOfAccountActivities($time)
+    $: activityTileDataList = searchActivities(getActivityTileDataList($queriedActivities), searchValue)
 
     function getActivityGroupTitleForTimestamp(time: Date): string {
         const dateString = getMonthYear(time)
         return dateString === getMonthYear(new Date()) ? localize('general.thisMonth') : dateString
     }
+
+    function getActivityTileDataList(activities: Activity[]): ActivityTileData[] {
+        return activities.map((activity, index) => {
+            const currentTitle = getActivityGroupTitleForTimestamp(activity.time)
+            const previousTitle =
+                activities[index - 1] && getActivityGroupTitleForTimestamp(activities[index - 1]?.time)
+            if (currentTitle !== previousTitle || index === 0) {
+                const amount = activities.filter(
+                    (activity) => getActivityGroupTitleForTimestamp(activity.time) === currentTitle
+                ).length
+
+                return { title: currentTitle, amount, activity }
+            } else {
+                return { title: undefined, amount: undefined, activity }
+            }
+        })
+    }
+
+    function searchActivities(activities: ActivityTileData[], searchValue: string): ActivityTileData[] {
+        if (!activities || !searchValue) {
+            return activities
+        }
+
+        const searchTerm = searchValue.toLowerCase()
+        const filteredActivities = activities?.filter((activity) => {
+            const title = localize(getActivityTileTitle(activity?.activity))?.toLowerCase()
+            const transactionId = activity?.activity?.transactionId?.toLowerCase()
+
+            const addressOrName =
+                activity.activity.subject.type === 'address'
+                    ? activity.activity.subject.address.toLowerCase()
```

Comment:
im getting a crash here, we might need to add some `?.` here and there to avoid all possible crashes

Reply:
the error is:
```
index.js:31 TypeError: Cannot read properties of undefined (reading 'toLowerCase')
    at ActivityList.svelte:66:1
    at Array.filter (<anonymous>)
    at searchActivities (ActivityList.svelte:60:1)
    at $$self.$$.update (ActivityList.svelte:30:1)
    at update (index.mjs:1205:1)
    at flush (index.mjs:1169:1)
```

Reply:
I get now a new crash 🙏🏼 
```
index.js:31 TypeError: Cannot read properties of undefined (reading 'includes')
    at ActivityList.svelte:77:1
    at Array.some (<anonymous>)
    at ActivityList.svelte:77:1
    at Array.filter (<anonymous>)
    at searchActivities (ActivityList.svelte:60:1)
    at $$self.$$.update (ActivityList.svelte:30:1)
    at update (index.mjs:1205:1)
    at flush (index.mjs:1169:1)
```